pkg/apis/networkoperator/v1/types.go

```diff
@@ -134,6 +137,21 @@ type OVNKubernetesConfig struct {
 // NetworkType describes the network plugin type to configure
 type NetworkType string
 
+// ProxyConfig defines the configuration knobs for kubeproxy
+// All of these are optional and have sensible defaults
+type ProxyConfig struct {
+	// The period that iptables rules are refreshed.
+	// Default: 30s
+	IptablesSyncPeriod string `json:"iptablesSyncPeriod,omitempty"`
+
+	// The address to "bind" on
+	// Defaults to 0.0.0.0
+	BindAddress string
+
+	// Any additional arguments to pass to the kubeproxy process
+	ProxyArguments map[string][]string
```

Comment:
Maybe the right way to handle kube-proxy configuration is to just *only* have ProxyArguments, and then the SDN node process can convert that into a [insert name of upstream kube-proxy config object] using the same functions kube-proxy itself does (so as to get defaults, etc) to extract the values of any fields it cares about.

In particular:
- As dcbw said, we need iptables minSyncPeriod too
- There's no good argument for exposing the iptables sync period fields but *not* exposing the proxy mode. (And likewise various other fields)
- The current system would let you, eg, set `BindAddress` to one value in this struct and then to a different value in the `ProxyArguments`

Reply:
I didn't think the proxymode was configurable - https://github.com/openshift/origin/blob/master/pkg/cmd/server/kubernetes/network/options/options.go#L60 ?

Reply:
That's the default value. The call to `cmdflags.Resolve` at the end then overrides the defaults using the values from `ProxyArguments`.

Reply:
Oh duh.